$MirOS: ports/lang/egcs/gcc4.4/patches/patch-gcc_config_gcc,v 1.2 2009/12/06 19:05:03 tg Exp $
--- gcc/config.gcc.orig	Fri Feb 18 21:39:27 2011
+++ gcc/config.gcc	Sun Dec  1 04:04:45 2013
@@ -522,6 +522,21 @@ case ${target} in
   default_use_cxa_atexit=yes
   use_gcc_tgmath=no
   ;;
+*-*-mirbsd*)
+  # don't build libgcc2 as PIC, multilib takes care of that on all targets
+  tmake_file="t-slibgcc-elf-ver t-mirbsd"
+  case ${enable_threads} in
+    yes|posix)
+      thread_file='posix'
+      tmake_file="${tmake_file} t-openbsd-thread"
+      ;;
+  esac
+
+  gas=yes
+  gnu_ld=yes
+
+  extra_parts="crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o"
+  ;;
 *-*-netbsd*)
   tmake_file="t-slibgcc-elf-ver t-libc-ok t-netbsd t-libgcc-pic"
   gas=yes
@@ -1053,6 +1068,11 @@ i[34567]86-*-freebsd*)
 x86_64-*-freebsd*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h ${fbsd_tm_file} i386/x86-64.h i386/freebsd.h i386/freebsd64.h"
 	tmake_file="${tmake_file} i386/t-crtstuff"
+	;;
+i?86-*-mirbsd*)
+	tmake_file="${tmake_file} i386/t-openbsd i386/t-crtstuff"
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h"
+	tm_file="${tm_file} mirbsd.h i386/mirbsd.h"
 	;;
 i[34567]86-*-netbsdelf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h netbsd.h netbsd-elf.h i386/netbsd-elf.h"
